vignettes/example.Rmd

```diff
@@ -50,21 +50,26 @@ sigma_true <- 2
 data_list <- list(x = rnorm(10, mean = mu_true, sd = sigma_true))
 ```
 
-For our example MCMC we will assume that we know the correct distribution of the data (i.e. we know the data is normally distributed), and we know that the mean is no smaller than -10 and no larger than 10, but otherwise the parameters of the distribution are unknown. Parameters within *drjacoby* are defined in dataframe format, where we specify minimum, maximum, and initial values of all parameters:
+Notice that data needs to be defined as a named list. For our example MCMC we will assume that we know the correct distribution of the data (i.e. we know the data is normally distributed), and we know that the mean is no smaller than -10 and no larger than 10, but otherwise the parameters of the distribution are unknown. Parameters within *drjacoby* are defined in dataframe format, where we specify minimum, maximum, and initial values of all parameters. The function `define_params()` makes it slightly easier to define this dataframe, but standard methods are also fine:
 
 ```{r}
 # define parameters dataframe
-df_params <- data.frame(name = c("mu", "sigma"),
-                        min = c(-10, 0),
-                        max = c(10, Inf),
-                        init = c(9, 10))
+df_params <- define_params(name = "mu", min = -10, max = 10, init = 9,
+                           name = "sigma", min = 0, max = Inf, init = 10)
 
 print(df_params)
 ```
 
 In this example we have one parameter (`mu`) that occupies a finite range [-10, 10], and another parameter (`sigma`) that can take any positive value. *drjacoby* deals with different parameter ranges using reparameterisation, which all occurs internally meaning we don't need to worry about these constraints affecting our inference.
 
-Next, we need a likelihood function. This **must** have exactly two input arguments: 1) a vector of parameters, 2) a vector of data, and these **must** be input **in that order**. It also **must** return a single value for the likelihood **in log space**. These constraints on the format of the likelihood function might seem a bit restrictive, but they are needed in order for *drjacoby* to know how to use the function internally. The issue of taking logs is particularly important, as the MCMC will still run even if we forget to take logs, but the results produced will be nonsense!
+Next, we need a likelihood function. This **must** have the following four input arguments, in this order:
+
+1. `params` - a named vector of parameter values
+2. `param_i` - an index for the current parameter being updated

```

Comment:
I've missed a lot of iteration here that I assume went into Nick's work - so apologies if my understanding is superficial. I was wondering whether it was possible to wrap `param_i` into a named element of `misc`, seeing as I guess for most users situations it won't be required?